tests/test_data_open_dataset.py

```diff
@@ -11,116 +10,122 @@
 
 
 class TestOpenDataset(unittest.TestCase):
+    def test_open_dataset_with_null_dataset_raises(self):
+        with self.assertRaises(ValueError):
+            open_dataset("")
 
-    def setUp(self):
-        self.patch_dataset_config_ret_val = {
-            "giops": {
-                "enabled": True,
-                "url": "tests/testdata/nemo_test.nc",
-                "grid_angle_file_url": "tests/testdata/nemo_grid_angle.nc",
-                "time_dim_units": "seconds since 1950-01-01 00:00:00",
-                "quantum": "day",
-                "name": "GIOPS",
-                "help": "help",
-                "attribution": "attrib",
-                "variables": {
-                    "votemper": {"name": "Temperature", "scale": [-5, 30], "units": "Kelvins", "equation": "votemper - 273.15"},
-                }
-            }
+    @patch.object(DatasetConfig, "_get_dataset_config")
+    def test_open_dataset_with_null_url_raises(self, patch_get_dataset_config):
+        patch_get_dataset_config.return_value = {
+            "giops": {"url": None, "variables": {}}
         }
+        config = DatasetConfig("giops")
 
-        self.patch_dataset_config_ret_val_sqlite = {
-            "giops": {
-                "enabled": True,
-                "url": "tests/testdata/databases/test-nemo.sqlite3",
-                "grid_angle_file_url": "tests/testdata/nemo_grid_angle.nc",
-                "time_dim_units": "seconds since 1950-01-01 00:00:00",
-                "quantum": "day",
-                "name": "GIOPS",
-                "help": "help",
-                "attribution": "attrib",
-                "variables": {
-                    "votemper": {"name": "Temperature", "scale": [-5, 30], "units": "Kelvins", "equation": "votemper - 273.15"},
-                }
-            }
-        }
+        with self.assertRaises(ValueError):
+            open_dataset(config)
 
-    def test_open_dataset_returns_nemo_object(self):
+    @patch.object(data.calculated, "CalculatedData")
+    def test_open_dataset_bad_dim_list_raises(self, patch_calculated_data):
+        with self.assertRaises(RuntimeError):
+            open_dataset("tests/testdata/does_not_exist.nc", meta_only=True)
 
-        with open_dataset('tests/testdata/nemo_test.nc') as ds:
-            self.assertTrue(isinstance(ds, Nemo))
+    @patch.object(DatasetConfig, "_get_dataset_config")
+    @patch.object(data.calculated, "CalculatedData")
+    def test_open_dataset_meta_only_returns_mercator_object(
+        self, patch_calculated_data, patch_get_dataset_config
+    ):
+        patch_get_dataset_config.return_value = {
+            "giops": {"url": "tests/testdata/mercator_test.nc", "variables": {}}
```

Comment:
If these mocks get too crazy/repetitive, feel free to create some fixtures in `tests/testdata`.

Reply:
Thanks., I'll keep that in mind. In this case I need just enough a config to get through to the real business of `open_dataset()`, so I feel like they are too small to deserve separate fixture files.